src/main/java/com/internship/backend/service/UserService.java

```diff
@@ -1,34 +1,100 @@
 package com.internship.backend.service;
 
+import com.internship.backend.dto.UserDTO;
 import com.internship.backend.exceptions.UserAlreadyExistsException;
 import com.internship.backend.exceptions.UserDoesNotExistException;
+import com.internship.backend.model.Authority;
 import com.internship.backend.model.Users;
+import com.internship.backend.repository.AuthorityRepository;
 import com.internship.backend.repository.UserRepository;
+import org.apache.commons.lang3.ObjectUtils;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.stereotype.Service;
 
+import java.util.HashSet;
+import java.util.Iterator;
 import java.util.List;
+import java.util.Set;
+
 
 @Service
 public class UserService {
 
     @Autowired
     private UserRepository userRepository;
 
+    @Autowired
+    private PasswordEncoder passwordEncoder;
+
+
+    @Autowired
+    private AuthorityRepository authorityRepository;
+
+    public Users register(Users user) throws UserAlreadyExistsException {
+        if (userRepository.findByUsername(user.getUsername()) != null) {
+            throw new UserAlreadyExistsException("Username already exists");
+        }
+        if (userRepository.findByEmail(user.getEmail()) != null) {
+            throw new UserAlreadyExistsException("Email already exists");
+        }
```

Comment:
It can be extracted in a dedicated method responsible for validating register data